sys/net/gnrc/network_layer/sixlowpan/frag/sfr/gnrc_sixlowpan_frag_sfr.c

```diff
@@ -0,0 +1,1655 @@
+        /* send ACK further down */
+        DEBUG("6lo sfr: forward ACK to (%s, %02x)\n",
+              gnrc_netif_addr_to_str(vrbe->super.src, vrbe->super.src_len,
+                                     addr_str), vrbe->super.tag);
+        _send_ack(vrbe->in_netif, vrbe->super.src, vrbe->super.src_len,
+                  &mock_base, hdr->bitmap);
+        if (IS_USED(MODULE_GNRC_SIXLOWPAN_FRAG_SFR_STATS)) {
+            _stats.acks.forwarded++;
+        }
+        if ((unaligned_get_u32(hdr->bitmap) == _full_bitmap.u32) ||
+            (unaligned_get_u32(hdr->bitmap) == _null_bitmap.u32)) {
+            if (CONFIG_GNRC_SIXLOWPAN_FRAG_RBUF_DEL_TIMER > 0) {
+                /* garbage-collect entry after CONFIG_GNRC_SIXLOWPAN_FRAG_RBUF_DEL_TIMER
+                 * microseconds */
+                vrbe->super.arrival = xtimer_now_usec() -
+                                      (CONFIG_GNRC_SIXLOWPAN_FRAG_VRB_TIMEOUT_US -
+                                       CONFIG_GNRC_SIXLOWPAN_FRAG_RBUF_DEL_TIMER);
+            }
+            else {
+                gnrc_sixlowpan_frag_vrb_rm(vrbe);
+            }
+        }
+        else {
+            vrbe->super.arrival = xtimer_now_usec();
+        }
+    }
+    else {
+        gnrc_sixlowpan_frag_fb_t *fbuf;
+
+        if ((fbuf = gnrc_sixlowpan_frag_fb_get_by_tag(hdr->base.tag)) != NULL) {
+            /* removing ACK timeout */
+            DEBUG("6lo sfr: cancelling ARQ timeout\n");
+            evtimer_del((evtimer_t *)(&_arq_timer),
+                        &fbuf->sfr.arq_timeout_event.event);
+            fbuf->sfr.arq_timeout_event.msg.content.ptr = NULL;
+            if ((unaligned_get_u32(hdr->bitmap) == _null_bitmap.u32)) {
+                DEBUG("6lo sfr: fragmentation canceled\n");
+                _retry_datagram(fbuf);
+            }
+            else {
+               _check_failed_frags(hdr, fbuf);
+            }
+        }
+        else {
+            DEBUG("6lo sfr: no VRB or fragmentation buffer found\n");
+        }
+    }
+    gnrc_pktbuf_release(pkt);
+}
+
+static int _forward_rfrag(gnrc_pktsnip_t *pkt, _generic_rb_entry_t *entry,
+                          uint16_t offset, unsigned page)
+{
+    gnrc_pktsnip_t *new = gnrc_netif_hdr_build(
+            NULL, 0,
+            entry->entry.base->dst, entry->entry.base->dst_len
+        );
+
+    assert(entry->type == _VRB);
+    /* restrict out_tag to value space of SFR, so that later RFRAG ACK can find
+     * it in reverse look-up */
+    entry->entry.vrb->out_tag &= UINT8_MAX;
+    DEBUG("6lo sfr: Forwarding to (%s, %u)\n",
+          gnrc_netif_addr_to_str(entry->entry.base->dst,
+                                 entry->entry.base->dst_len, addr_str),
+          entry->entry.vrb->out_tag);
+    if (new == NULL) {
+        DEBUG("6lo sfr: Unable to forward fragment, "
+              "packet buffer full\n");
+        gnrc_pktbuf_release(pkt);
+        return -ENOMEM;
+    }
+    else {
```

Comment:
```suggestion
```
since you `return` in the case before, you can save yourself a level of indentation here.